services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/LinkOperationsBolt.java

```diff
@@ -0,0 +1,166 @@
+            List result = null;
+            if (request instanceof GetLinksRequest) {
+                result = getAllLinks(session);
+            } else {
+                LOGGER.warn("Received unsupported Links request: {}", correlationId);
+            }
+
+            LOGGER.debug("Processed read request with correlationId {}", correlationId);
+            getOutput().emit(Lists.newArrayList(result, correlationId));
+        }
+
+    }
+
+    private List<IslInfoData> getAllLinks(Session session) {
+        LOGGER.debug("Getting ISLs...");
+        String q =
+                "MATCH (:switch)-[isl:isl]->(:switch) "
+                        + "RETURN "
+                        + "isl.src_switch as src_switch, "
+                        + "isl.src_port as src_port, "
+                        + "isl.dst_switch as dst_switch, "
+                        + "isl.dst_port as dst_port, "
+                        + "isl.speed as speed, "
+                        + "isl.max_bandwidth as max_bandwidth, "
+                        + "isl.latency as latency, "
+                        + "isl.available_bandwidth as available_bandwidth, "
+                        + "isl.status as status";
+
+        StatementResult queryResults = session.run(q);
+        List<IslInfoData> results = new ArrayList<>();
+        for (Record record : queryResults.list()) {
+            LOGGER.debug("Reading record... {}", record);
+            // max_bandwidth not used in IslInfoData
+            PathNode src = new PathNode();
+            src.setSwitchId(record.get("src_switch").asString());
+            src.setPortNo(parseIntValue(record.get("src_port")));
+            src.setSegLatency(parseIntValue(record.get("latency")));
+            PathNode dst = new PathNode();
+            dst.setSwitchId(record.get("dst_switch").asString());
+            dst.setPortNo(parseIntValue(record.get("dst_port")));
+            dst.setSegLatency(parseIntValue(record.get("latency")));
+
+            List<PathNode> pathNodes = new ArrayList<>();
+            pathNodes.add(src);
+            pathNodes.add(dst);
+
+            IslChangeType state = getStatus(record.get("status").asString());
+            IslInfoData isl = new IslInfoData(
+                    parseIntValue(record.get("latency")),
+                    pathNodes,
+                    parseIntValue(record.get("speed")),
+                    state,
+                    parseIntValue(record.get("available_bandwidth"))
+            );
+            isl.setTimestamp(System.currentTimeMillis());
+
+            results.add(isl);
+            LOGGER.info("Result ISL: {}", isl);
+
+        }
+        return results;
+    }
+
+    @Override
+    public void declareOutputFields(OutputFieldsDeclarer declarer) {
+        declarer.declare(new Fields("response", "correlationId"));
+    }
+
+    @Override
+    public void cleanup() {
+        driver.close();
+    }
+
+    private int parseIntValue(Value value) {
```

Comment:
It can't handle numeric values stored as strings. There is a lot such objects in DB.